playgrounds/eav-checkbox-playground.R

```diff
@@ -0,0 +1,559 @@
+
+# ---- load-packages -----------------------------------------------------------
+library("magrittr")
+requireNamespace("dplyr")
+requireNamespace("readr")
+requireNamespace("testit")
+
+# ---- declare-globals ---------------------------------------------------------
+# as projects are only containing test data, okay to commit URL and tokens
+# to Github in private repo
+# remove if we ever submit a PR to the main project
+redcap_uri <- "https://redcap.hbcd.cmig-ucsd.org/api/"
+
+# token      <- "53320377CD34F60392870F5F6C20953B" # longitudinal & repeating
+# token      <- "778FAF179F3DBEDBC1A3CA1CAB2B36EF" # repeating
+# token      <- "831D17A42D03BC448A7B1A08CDE19AD9" # longitudinal
+token      <- "B77827EB144D9B7D74F01E932C9F0B29" # neither
+
+fields  <- NULL
+# fields   <- c(
+#   # "record_id",
+#   "form_1_check_001"
+# )
+forms   <- NULL
+# forms   <- c(
+#   "form_1",
+#   # "form_2",
+#   "form_3"
+#   # "form_4"
+# )
+events  <- NULL
+# events  <- c(
+#   "event_1_arm_1"
+#   # "event_3_arm_1"
+# )
+records <- NULL
+
+blank_for_gray_form_status <- TRUE
+.complete_value_for_untouched_forms <- dplyr::if_else(
+  blank_for_gray_form_status,
+  NA_character_,
+  as.character(REDCapR::constant("form_incomplete"))
+)
+
+# NOTE: I think that an additional function parameter to specify how to handle
+#   untouched checkbox items might make sense. But we can also make it dependent
+#   on `blank_for_gray_form_status` if you prefer that. I just think some people
+#   might want to make different selections for either.
+blank_for_untouched_checkbox <- TRUE
+
+# ---- load-data ---------------------------------------------------------------
+# View(REDCapR::redcap_variables(redcap_uri, token)$data)
+system.time({
+  meta <- REDCapR:::redcap_metadata_internal(
+    redcap_uri,
+    token
+  )
+
+  ds_metadata          <- meta$d_variable
+  ds_metadata_plumbing <- ds_metadata |>
+    dplyr::filter(
+      plumbing
+    )
+
+  # only include selected fields/forms
+  if (!is.null(fields) & !is.null(forms)) {
+    ds_metadata <- ds_metadata |>
+      dplyr::filter(
+        field_name_base %in% fields | form_name %in% forms
+      )
+  } else if (!is.null(fields)) {
+    ds_metadata <- ds_metadata |>
+      dplyr::filter(
+        field_name_base %in% fields
+      )
+  } else if (!is.null(forms)) {
+    ds_metadata <- ds_metadata |>
+      dplyr::filter(
+        form_name %in% forms
+      )
+  }
+
+  col_types <- REDCapR::redcap_metadata_coltypes(
+    redcap_uri,
+    token,
+    print_col_types_to_console = FALSE
+  )
+
+  # only include selected fields/forms
+  col_types[[1]] <- col_types[[1]][ds_metadata$field_name]
+
+  ds_eav <- REDCapR:::redcap_read_eav_oneshot(
+    redcap_uri,
+    token,
+    fields               = fields,
+    forms                = forms,
+    events               = events,
+    records              = records,
+    export_survey_fields = TRUE
+  )$data
+})
```

Comment:
From my understanding `redcap_read` reads the table and expand choices if there are many for a single question, but `eva` has one `value` column to store the value, like a `tidyr::pivot_longer` result?

Reply:
See above for my description of "standard" (`redcap_read`; which uses `redcap_read_oneshot` as a utility function underneath) and "EAV" (`redcap_read_eav_oneshot`)